packages/compiler/test/parser.test.ts

```diff
@@ -713,6 +714,26 @@ describe("compiler: parser", () => {
         { strict: true }
       );
     });
+
+    describe("ensure directives and decorators are applied to leaf node", () => {
+      parseEach([
+        [
+          `@doc("foo")\n#suppress "foo"\nnamespace Foo.Bar {}`,
+          (node) => {
+            const fooNs = node.statements[0];
+            assert(fooNs.kind === SyntaxKind.NamespaceStatement);
+            const barNs = (fooNs as any).statements;
```

Comment:
this as shouldn't be needed either

Reply:
It's not needed, but it makes the logic more annoying because this is a union type.

Reply:
ah right forgot